pkg/build/api/types.go

```diff
@@ -260,32 +260,63 @@ const (
 	// output is an invalid reference.
 	StatusReasonInvalidOutputReference StatusReason = "InvalidOutputReference"
+	StatusMessageInvalidOutputRef                   = "Output image could not be resolved"
 
 	// StatusReasonCancelBuildFailed is an error condition when cancelling a build
 	// fails.
-	StatusReasonCancelBuildFailed StatusReason = "CancelBuildFailed"
+	StatusReasonCancelBuildFailed  StatusReason = "CancelBuildFailed"
+	StatusMessageCancelBuildFailed              = "Failed to cancel build"
 
 	// StatusReasonBuildPodDeleted is an error condition when the build pod is
 	// deleted before build completion.
-	StatusReasonBuildPodDeleted StatusReason = "BuildPodDeleted"
+	StatusReasonBuildPodDeleted  StatusReason = "BuildPodDeleted"
+	StatusMessageBuildPodDeleted              = "The pod for this build was deleted before the build completed"
 
 	// StatusReasonExceededRetryTimeout is an error condition when the build has
 	// not completed and retrying the build times out.
-	StatusReasonExceededRetryTimeout StatusReason = "ExceededRetryTimeout"
+	StatusReasonExceededRetryTimeout  StatusReason = "ExceededRetryTimeout"
+	StatusMessageExceededRetryTimeout              = "Build did not complete and retrying timed out"
 
 	// StatusReasonMissingPushSecret indicates that the build is missing required
 	// secret for pushing the output image.
 	// The build will stay in the pending state until the secret is created, or the build times out.
-	StatusReasonMissingPushSecret StatusReason = "MissingPushSecret"
+	StatusReasonMissingPushSecret  StatusReason = "MissingPushSecret"
+	StatusMessageMissingPushSecret              = "Missing push secret"
+
+	// StatusReasonPostCommitHookFailed indicates the post-commit hook failed.
+	StatusReasonPostCommitHookFailed  StatusReason = "PostCommitHookFailed"
+	StatusMessagePostCommitHookFailed              = "Build failed because of post commit hook"
+
+	// StatusReasonPushImageToRegistryFailed indicates that an image failed to be
+	// pushed to the registry.
+	StatusReasonPushImageToRegistryFailed  StatusReason = "PushImageToRegistryFailed"
+	StatusMessagePushImageToRegistryFailed              = "Failed to push the image to the registry"
+
+	// StatusReasonPullBuilderImageFailed indicates that we failed to pull the
+	// builder image.
+	StatusReasonPullBuilderImageFailed  StatusReason = "PullBuilderImageFailed"
+	StatusMessagePullBuilderImageFailed              = "Failed pulling builder image"
+
+	// StatusReasonFetchSourceFailed indicates that fetching the source of the
+	// build has failed.
+	StatusReasonFetchSourceFailed  StatusReason = "FetchSourceFailed"
+	StatusMessageFetchSourceFailed              = "Failed to fetch the input source"
+
+	// StatusReasonCancelledBuild indicates that the build was cancelled by the
+	// user.
+	StatusReasonCancelledBuild  StatusReason = "CancelledBuild"
+	StatusMessageCancelledBuild              = "The build was cancelled by the user"
```

Comment:
These messages aren't part of the public API.  Move these messages to someplace appropriate.